### ISSUE TITLE
Always bidirectional RNNs

### ISSUE THREAD

**Issue by ArtemiyFirsov** [SeanNaren/deepspeech.pytorch#102]
Hi!

Noticed that in this peace of code in "train.py" bidirectional param is always set to true. 

`model = DeepSpeech(rnn_hidden_size=args.hidden_size,
                       nb_layers=args.hidden_layers,
                       labels=labels,
                       rnn_type=supported_rnns[rnn_type],
                       audio_conf=audio_conf,
                       bidirectional=True)
`

I guess it should be in args.

**Comment by SeanNaren** [comment]
This is intentional, at the current stage uni-directional RNNs are not supported correctly and are substantially worse than bi-directional. The inclusion of uni-directional RNNs depends on #46 !

**Comment by ArtemiyFirsov** [comment]
Ok, thank you! That would be a super feature. BTW, is there any docs, that describe what features are already implemented? Like sortagrad/batch norm/LM training/etc...?

**Comment by SeanNaren** [comment]
Not a bad idea, I'll create a ticket for a support feature list in the main README that I'll update as we progress :)

**Comment by dlmacedo** [comment]
This will GREAT!

**Comment by ArtemiyFirsov** [comment]
Super! I would suggest to start with features below, they are described briefly - need to decorate those:
 
1. Configurable NN structure, i.e. configurable number of first convolution layers, configurable RNN layer(s) - GRU/LSTM/... + bidir or forward, configurable number of last layers. Also I think there should be a possibiity to set number of neurons in a layer.
2. Dropout technique with ability to set percentage of dropout
3. Signal jittering set with boolean flag.
4. LM model training with different methods + differnet sources of phrases for training
5. Maybe dynamic batch size to fit GPU memory automatically. Not sure if it is possible, just an idea.
6. Ordering train set by audio length for optimization reasons.
7. Striding - do not quite understand this concept from article, but it also is used for optimization
8. Synthesize audio using noise - so, need some noise sources + method for noise superposition
9. Speaker adaptation for small datasets
10. CV for parameters tuning
11. Ability to train ensemble of RNNs 
12. BatchNorm
13. SortaGrad

Maybe smth else?)
